src/main.rs

```diff
@@ -0,0 +1,189 @@
+use fold::{
+    interner::{InMemoryInternerHolder, InternerHolderLike},
+    ortho::Ortho,
+    ortho_database::{InMemoryOrthoDatabase, OrthoDatabaseLike},
+    FoldError,
+};
+use std::collections::VecDeque;
+use std::fs;
+use std::path::PathBuf;
+
+/// Simple in-memory queue implementation for the worker loop
+struct SimpleQueue {
+    items: VecDeque<Ortho>,
+}
+
+impl SimpleQueue {
+    fn new() -> Self {
+        SimpleQueue {
+            items: VecDeque::new(),
+        }
+    }
+    
+    fn into_inner(self) -> VecDeque<Ortho> {
+        self.items
+    }
+}
+
+impl fold::queue::QueueProducerLike for SimpleQueue {
+    fn push_many(&mut self, items: Vec<Ortho>) -> Result<(), FoldError> {
+        self.items.extend(items);
+        Ok(())
+    }
+}
+
+impl fold::queue::QueueLenLike for SimpleQueue {
+    fn len(&mut self) -> Result<usize, FoldError> {
+        Ok(self.items.len())
+    }
+}
+
+fn main() -> Result<(), FoldError> {
+    let state_dir = std::env::var("FOLD_STATE_DIR").unwrap_or_else(|_| "./fold_state".to_string());
+    let input_dir = PathBuf::from(&state_dir).join("input");
+
+    if !input_dir.exists() {
+        eprintln!("Error: Input directory does not exist: {:?}", input_dir);
+        eprintln!("Run stage.sh to create input files first.");
+        return Ok(());
+    }
+
+    // Collect and sort input files
+    let mut input_files: Vec<PathBuf> = fs::read_dir(&input_dir)
+        .map_err(|e| FoldError::Io(e))?
+        .filter_map(|entry| {
+            let entry = entry.ok()?;
+            let path = entry.path();
+            if path.is_file() && path.extension().and_then(|s| s.to_str()) == Some("txt") {
+                Some(path)
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    input_files.sort();
+
+    if input_files.is_empty() {
+        println!("No input files found in {:?}", input_dir);
+        return Ok(());
+    }
+
+    println!("Found {} input files", input_files.len());
+
+    // Initialize interner holder and database
+    let mut interner_holder = InMemoryInternerHolder::new()?;
+    let mut db = InMemoryOrthoDatabase::new();
```

Comment:
Remove the ortho database. Only store the optimal ortho. Check each ortho for optimality as it is produced.